src/transformers/pipelines/video_classification.py

```diff
@@ -0,0 +1,113 @@
+from io import BytesIO
+from typing import List, Union
+
+import requests
+
+from ..utils import add_end_docstrings, is_decord_available, is_torch_available, logging, requires_backends
+from .base import PIPELINE_INIT_ARGS, Pipeline
+
+
+if is_decord_available():
+    import numpy as np
+
+    from decord import VideoReader, cpu
+
+
+if is_torch_available():
+    from ..models.auto.modeling_auto import MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING
+
+logger = logging.get_logger(__name__)
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class VideoClassificationPipeline(Pipeline):
+    """
+    Video classification pipeline using any `AutoModelForVideoClassification`. This pipeline predicts the class of a
+    video.
+
+    This video classification pipeline can currently be loaded from [`pipeline`] using the following task identifier:
+    `"video-classification"`.
+
+    See the list of available models on
+    [huggingface.co/models](https://huggingface.co/models?filter=video-classification).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        requires_backends(self, "decord")
+        self.check_model_type(MODEL_FOR_VIDEO_CLASSIFICATION_MAPPING)
+
+        self.frame_sampling_rate = kwargs.pop("frame_sample_rate", 4)
+        self.num_frames = self.model.config.num_frames
```

Comment:
We should use `_sanitize_parameters` for `frame_sampling_rate`, we cannot rely on `self` for parameters because we always need to be able to know which values to use.

```python
pipe = pipeline(..., arg=1)
out = pipe(..) # Using arg=1
out = pipe(arg=2) # Using arg=2
out = pipe(...) # Using arg=1 again.
```

This is the reason for this weird `_sanitize_parameters` thing. To have a single location that handles that complexity.

`self.num_frames` seems to be used only in `preprocess` and should be defined only there. (without self).
Keep in mind that `preprocess` might be working on a different thread than main thread.